creator-node/src/services/stateMachineManager/stateReconciliation/updateReplicaSet.jobProcessor.ts

```diff
@@ -632,46 +637,65 @@ type CanReconfigParams = {
   userId: number
   logger: Logger
 }
+type CanReconfigReturnValue = {
+  canReconfig: boolean
+  chainPrimarySpId?: number
+  chainSecondarySpIds?: number[]
+}
 const _canReconfig = async ({
   libs,
   oldPrimarySpId,
   oldSecondary1SpId,
   oldSecondary2SpId,
   userId,
   logger
-}: CanReconfigParams): Promise<boolean> => {
-  // If any error occurs in determining if a reconfig event can happen, default to issuing
-  // a reconfig event anyway just to prevent users from keeping an unhealthy replica set
-  let canReconfig = true
+}: CanReconfigParams): Promise<CanReconfigReturnValue> => {
   try {
-    const {
-      primaryId: currentPrimarySpId,
-      secondaryIds: currentSecondarySpIds
-    } = await libs.contracts.UserReplicaSetManagerClient.getUserReplicaSet(
-      userId
-    )
+    const { primaryId: chainPrimarySpId, secondaryIds: chainSecondarySpIds } =
+      await libs.contracts.UserReplicaSetManagerClient.getUserReplicaSet(userId)
 
     if (
-      !currentPrimarySpId ||
-      !currentSecondarySpIds ||
-      currentSecondarySpIds.length < 2
+      !chainPrimarySpId ||
+      !chainSecondarySpIds ||
+      chainSecondarySpIds.length < 2
     ) {
       throw new Error(
-        `Could not get current replica set: currentPrimarySpId=${currentPrimarySpId} currentSecondarySpIds=${currentSecondarySpIds}`
+        `Could not get current replica set: chainPrimarySpId=${chainPrimarySpId} chainSecondarySpIds=${JSON.stringify(
+          chainSecondarySpIds || []
+        )}`
       )
     }
 
-    canReconfig =
-      currentPrimarySpId === oldPrimarySpId &&
-      currentSecondarySpIds[0] === oldSecondary1SpId &&
-      currentSecondarySpIds[1] === oldSecondary2SpId
+    // Reconfig is necessary if endpoint doesn't exist in mapping because this means the node was deregistered
+    const isAnyNodeInReplicaSetDeregistered =
+      !oldPrimarySpId || !oldSecondary1SpId || !oldSecondary2SpId
+    if (isAnyNodeInReplicaSetDeregistered) {
+      return {
+        canReconfig: true,
+        chainPrimarySpId,
+        chainSecondarySpIds
+      }
+    }
+
+    // Reconfig should only happen when the replica set that triggered the reconfig matches the chain
+    const isReplicaSetCurrent =
+      chainPrimarySpId === oldPrimarySpId &&
+      chainSecondarySpIds[0] === oldSecondary1SpId &&
+      chainSecondarySpIds[1] === oldSecondary2SpId
+    return {
+      canReconfig: isReplicaSetCurrent
+    }
   } catch (e: any) {
     logger.error(
       `[_issueUpdateReplicaSetOp] error in _canReconfig. : ${e.message}`
     )
   }
 
-  return canReconfig
+  // If any error occurs in determining if a reconfig event can happen, default to issuing
+  // a reconfig event anyway just to prevent users from keeping an unhealthy replica set
+  return {
```

Comment:
is this something you saw in testing? the uncaught error scenario?
can you think of any risks to doing this?
cc @dmanjunath

Reply:
I don't think so -- haven't observed it myself. the thinking here is to only short-circuit reconfigs when we know for sure they won't work, so that way we don't have to be too eager to short-circuit since we can let the chain call catch anything else

Reply:
this is actually how it was before the comment just got moved down. i think the only way we could get here is POA call failed cause gateway down or incorrectly formatted input. 

our thought process here is to send the update request to chain anyway is that if the old rset doesn't match the chain would reject

Reply:
oh yeah duh, agreed